client/src/components/generic/TextInputComponent/TextInput.tsx

```diff
@@ -0,0 +1,26 @@
+import React from "react"
+
+type TextInputProps = React.InputHTMLAttributes<HTMLInputElement> & {
+  success?: boolean
+  error?: boolean
+}
+
+const TextInput = ({ success, error, ...props }: TextInputProps) => {
```

Comment:
this is kinda sus, Its probably ok for the success state because we don't have a built in html property for that but for errors id still style it using the `error:` selector